src/components/TextInput/TextInput.tsx

```diff
@@ -243,23 +251,35 @@ const TextInput = ({
           debouncedChange(e);
         }}
         multiLineIsResizable={multiLineIsResizable}
-        ref={inputRef}
+        ref={internalInputRef}
         {...inputProps}
       />
-      {onClear && (
-        <StyledIconContainer onClick={handleClear} {...iconContainerProps}>
+      {(onClear || clearable) && (
+        <StyledIconContainer
+          onClick={
+            onClear
+              ? (e: SyntheticEvent<Element, Event>) => handleClear(e)
+              : () => {
+                  if (internalInputRef?.current?.value) {
+                    setInternalValue('');
+                    internalInputRef.current.value = '';
+                  }
+                }
+          }
+          {...iconContainerProps}
+        >
           <Icon path={mdiClose} size="1em" />
         </StyledIconContainer>
       )}
-      {showCharacterCount && maxLength && (
+      {showCharacterCount && (
         <StyledCharacterCount
           ref={characterCountRef}
           errorMessage={errorMessage}
           isValid={isValid}
-          textIsTooLong={(internalValue as string).length > maxLength}
+          textIsTooLong={maxLength ? (internalValue as string).length > maxLength : false}
           {...characterCountProps}
         >
-          {(internalValue as string).length} / {maxLength}
+          {(internalValue as string).length} {maxLength ? `/ ${maxLength}` : null}
```

Comment:
Just since maxLength is a number and could be 0 (not sure why a textinput would have 0 as a max length but a user might)
```suggestion
          {(internalValue as string).length} {maxLength !== undefined ? `/ ${maxLength}` : null}
```